### ISSUE TITLE
Refactor duplicate `IndexOutput` struct

### ISSUE THREAD

**Issue by leaf-potato** [GreptimeTeam/greptimedb#4586]
### What type of enhancement is this?

Refactor

### What does the enhancement do?

When I read the relevant code of the Mito engine, I found that two identical structures are defined:
https://github.com/GreptimeTeam/greptimedb/blob/975b8c69e5307c61b17c498f016e574c04bbcfc9/src/mito2/src/sst/index.rs#L68-L75
https://github.com/GreptimeTeam/greptimedb/blob/975b8c69e5307c61b17c498f016e574c04bbcfc9/src/mito2/src/sst/index.rs#L57-L64

Perhaps defining a base structure is sufficient. 
### Implementation challenges

```rust
#[derive(Debug, Clone, Default)]
pub struct IndexBaseOutput {
    /// Size of the index.
    pub index_size: ByteCount,
    /// Number of rows in the index.
    pub row_count: RowCount,
    /// Available columns in the index.
    pub columns: Vec<ColumnId>,
}

pub type InvertedIndexOutput = IndexBaseOutput;
pub type FulltextIndexOutput = IndexBaseOutput;
```
1. Define `IndexBaseOutput` struct.
2. type `IndexBaseOutput` to `InvertedIndexOutput` and `FulltextIndexOutput`.

In the future, if `InvertedIndexOutput` or `FulltextIndexOutput` has non-public fields, it can be modified like this:
```rust
struct InvertedIndexOutput {
    base_output: IndexBaseOutput,
    .....
}
```

**Comment by evenyag** [comment]
@zhongzc What do you think?

**Comment by zhongzc** [comment]
I have no objection, you are free to change it this way. My original idea was that these two indexes have the opportunity to present different `Output`, and it would be more convenient to separate them in case fields are added later.

**Comment by leaf-potato** [comment]
> I have no objection, you are free to change it this way.

Ok, I'll submit a PR later.

**Comment by evenyag** [comment]
Closed by #4592
